src/osd/PrimaryLogPG.cc

```diff
@@ -9276,7 +9276,9 @@ void PrimaryLogPG::finish_promote(int r, CopyResults *results,
   OpContextUPtr tctx =  simple_opc_create(obc);
   tctx->at_version = get_next_version();
 
-  ++tctx->delta_stats.num_objects;
+  if (!obc->obs.oi.has_manifest()) {
+    ++tctx->delta_stats.num_objects;
+  }
```

Comment:
I think it would make more sense to increase num_object, but perhaps also add a num_manifest counter.  We definitely want to see how many manifest redirects we have, and i think it'll make more sense to keep num_object inclusive (instead of adding num_object + num_manifest to get the actual object count).

Reply:
(We could add num_manifest accounting in a separate PR; no need to hold this one up)

Reply:
I will make a new PR about num_manifest accounting as your comments.